File: DarkOrbit 10.0/Net/netty/commands/PriceModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class PriceModule
    {
        public const short ID = 10487;

        public const short URIDIUM = 1;
        public const short CREDITS = 0;

        public int mAmount = 0;
        public short mType = 0;

        public PriceModule(short pType, int pAmount)
        {
            this.mType = pType;
            this.mAmount = pAmount;
        }

        public byte[] write()
        {
            var param1 = new ByteArray(ID);
            param1.writeShort(mType);
            param1.writeInt(mAmount >> 14 | mAmount << 18);
            return param1.Message.ToArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/PriceModule.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

import simulator.netty.ServerCommand;

/**
 Created by Pedro on 30-03-2015.
 */
public class PriceModule
        implements ServerCommand {

    public static short URIDIUM = 1;
    public static short CREDITS = 0;

    public static int ID = 7671;

    public int   mAmount = 0;
    public short mType   = 0;

    public PriceModule(short pType, int pAmount) {
        this.mType = pType;
        this.mAmount = pAmount;
    }

    public void write(DataOutputStream param1) {
        try {
            param1.writeShort(ID);
            this.writeInternal(param1);
        } catch (IOException e) {
        }
    }

    protected void writeInternal(DataOutputStream param1) {
        try {
            param1.writeShort(20005);
            param1.writeInt(this.mAmount >>> 15 | this.mAmount << 17);
            param1.writeShort(this.mType);
        } catch (IOException e) {
        }
    }
}
